cime_config/hist_config.py

```diff
@@ -0,0 +1,1126 @@
+    def volume(self):
+        """Return the volume for this HistoryVolConfig object"""
+        return self.__volume
+
+    @property
+    def precision(self):
+        """Return the precision property for this HistoryVolConfig object"""
+        return self.__precision
+
+    def set_precision(self, prec, pobj, logger):
+        """Modify the precision property of this HistoryVolConfig object.
+        Return True if <prec> is a recognized precision"""
+        if prec.upper() in _OUT_PRECS:
+            self.__precision = prec
+            self.__precision_set = True
+            if logger.getEffectiveLevel() <= logging.DEBUG:
+                ctx = context_string(pobj)
+                logger.debug(f"Setting precision to '{prec}'{ctx}")
+            # end if
+            return True
+        # end if
+        emsg = f"Attempt to set unrecognized precision, '{prec}'"
+        pobj.add_syntax_err(emsg)
+        return False
+
+    @property
+    def max_frames(self):
+        """Return the max_frames property for this HistoryVolConfig object"""
+        return self.__max_frames
+
+    @property
+    def write_nstep0(self):
+        """Return the write_nstep0 property for this HistoryVolConfig object"""
+        return self.__write_nstep0
+
+    def set_max_frames(self, nframes, pobj, logger):
+        """Modify the max_frames property of this HistoryVolConfig object.
+        Return True if <nframes> is a valid setting."""
+        nframes_ok = True
+        nframes_i, _ = _is_integer(nframes)
+        nframes_ok = nframes_i and (nframes > 0)
+        if nframes_ok:
+            self.__max_frames = nframes_i
+            self.__max_frames_set = True
+            if logger.getEffectiveLevel() <= logging.DEBUG:
+                ctx = context_string(pobj)
+                logger.debug(f"Setting max frames to '{nframes}'{ctx}")
+            # end if
+        else:
+            emsg = f"Attempt to set max frames to '{nframes}', must be a positive integer"
+            pobj.add_syntax_err(emsg)
+        # end if
+        return nframes_ok
+
+    def set_write_nstep0(self, write_nstep0, pobj, logger):
+        """Modify the write_nstep0 property of this HistoryVolConfig object.
+        Return True if valid"""
+        true_values = ["true", "t", ".true."]
+        false_values = ["false", "f", ".false."]
+        nstep0_ok = True
+        if write_nstep0.lower() in true_values:
+            self.__write_nstep0 = ".true."
+        elif write_nstep0.lower() in false_values:
+            self.__write_nstep0 = ".false."
+        else:
+            nstep0_ok = False
+            emsg = f"Attempt to set write_nstep0 to '{write_nstep0}', must be true or false"
+            pobj.add_syntax_err(emsg)
+        # end if
+        if nstep0_ok and logger.getEffectiveLevel() <= logging.DEBUG:
+            ctx = context_string(pobj)
+            logger.debug(f"Setting write_nstep0 to '{self.__write_nstep0}'{ctx}")
+        # end if
+        return nstep0_ok
```

Comment:
```suggestion
        if write_nstep0.lower() in __TRUE_VALUES:
            self.__write_nstep0 = ".true."
        elif write_nstep0.lower() in __FALSE_VALUES:
            self.__write_nstep0 = ".false."
        else:
            emsg = f"Attempt to set write_nstep0 to '{write_nstep0}', must be one of {__TRUE_VALUES | __FALSE_VALUES}"
            pobj.add_syntax_err(emsg)
            return False
        # end if
        if logger.getEffectiveLevel() <= logging.DEBUG:
            ctx = context_string(pobj)
            logger.debug(f"Setting write_nstep0 to '{self.__write_nstep0}'{ctx}")
        # end if
        return True
```
I think this is a little simpler with the explicit return statements.

Reply:
updated.